src/module/group.rs

```diff
@@ -0,0 +1,146 @@
+        })
+    }
+
+    /// Register a collector to the group.
+    ///
+    /// ```
+    /// group
+    ///     .register(
+    ///         Box::new(FirstCollector::new()?,
+    ///         Box::<FirstEvent>::default()))?,
+    ///     )?
+    ///     .register(
+    ///         Box::new(SecondCollector::new()?,
+    ///         Box::<SecondEvent>::default()))?,
+    ///     )?;
+    /// ```
+    pub(crate) fn register(
+        &mut self,
+        id: ModuleId,
+        collector: Box<dyn Collector>,
+        section_factory: Box<dyn EventSectionFactory>,
+    ) -> Result<&mut Self> {
+        // Ensure uniqueness of the collector name. This is important as their
+        // name is used as a key.
+        if self.modules.get(&id).is_some() {
+            bail!(
+                "Could not insert collector '{}'; name already registered",
+                id,
+            );
+        }
+
+        match &mut self.section_factories {
+            Some(factories) => factories.insert(id, section_factory),
+            None => bail!("Section factories map no found"),
+        };
+
+        self.modules.insert(id, collector);
+        Ok(self)
+    }
+
+    /// Start the event retrieval in the factory.
+    pub(crate) fn start_factory(&mut self) -> Result<()> {
+        let section_factories = match self.section_factories.take() {
+            Some(factories) => factories,
+            None => bail!("No section factory found, aborting"),
+        };
+
+        self.factory.start(section_factories)
+    }
+
+    /// Sometimes we need to perform actions on factories at a higher level.
+    /// It's a bit of an hack for now, it would be good to remove it. One option
+    /// would be to move the core EventSection and their factories into modules
+    /// directly (using mandatory modules). This should not affect the module
+    /// API though, so it should be fine as-is for now.
+    #[cfg(not(test))]
+    pub(crate) fn get_section_factory<T: EventSectionFactory + 'static>(
+        &mut self,
+        id: ModuleId,
+    ) -> Option<&mut T> {
+        match self.section_factories.as_mut() {
+            Some(section_factories) => match section_factories.get_mut(&id) {
+                Some(module) => module.as_any_mut().downcast_mut::<T>(),
+                None => None,
+            },
+            None => None,
+        }
+    }
+}
+
+pub(crate) fn get_modules(factory: Box<dyn EventFactory>) -> Result<Group> {
+    let mut group = Group::new(factory)?;
+
+    // Register all collectors here.
+    group
+        .register(
+            ModuleId::SkbTracking,
+            Box::new(SkbTrackingCollector::new()?),
+            Box::<SkbTrackingEvent>::default(),
+        )?
+        .register(
+            ModuleId::Skb,
+            Box::new(SkbCollector::new()?),
+            Box::<SkbEvent>::default(),
+        )?
+        .register(
+            ModuleId::Ovs,
+            Box::new(OvsCollector::new()?),
+            Box::<OvsEvent>::default(),
```

Comment:
Just double-checking here: We are not forcing the same struct to implement `EventType` and `EventSectionFactory` right?